src/nsolid.h

```diff
@@ -961,7 +962,45 @@ class NODE_EXTERN Snapshot {
                           Cb&& cb,
                           Data&&... data);
 
+  /**
+   * @brief activates the heap profiler tracking objects from a specific JS
+   * thread.
+   *
+   * @param envinst SharedEnvInst of thread to take the snapshot from.
+   * @param redacted to get redacted strings in the heap snapshot.
+   * @param data variable number of arguments to be propagated to the callback.
+   * @return NSOLID_E_SUCCESS in case of success or a different NSOLID_E_
+   * error value otherwise.
+   */
+  template <typename... Data>
+  static int StartTrackingHeapObjects(SharedEnvInst envinst,
+                                      bool trackAllocations,
+                                      Data&&... data);
+
```

Comment:
I'd change the interface a bit and make it more like the `CpuProfile`:
```suggestion
  template <typename Cb, typename... Data>
  static int TrackHeapObjects(SharedEnvInst envinst,
                              bool trackAllocations,
                              uint64_t duration,
                              Cb&& cb,
                              Data&&... data);

```
Then have a
```c++
static int StopTrackingHeapObjects(SharedEnvInst envinst);
```

Reply:
Fixed with the timers support